r2/service_metrics.go

```diff
@@ -78,7 +79,7 @@ func (sm *serviceMetrics) recordMetric(path, method, namespace string, d time.Du
 }
 
 func newServiceMetrics(scope tally.Scope) serviceMetrics {
-	return serviceMetrics{scope: scope}
+	return serviceMetrics{scope: scope, metrics: map[serviceMetricID]serviceMetric{}}
```

Comment:
nit: `metrics: make(map[serviceMetricID]serviceMetric)` is a bit more idiomatic in Go.